cmd/common/registry.go

```diff
@@ -0,0 +1,377 @@
+						},
+					))
+					break
+				}
+				fe.Output(formatter.FormatHeader())
+				parser.SetEventCallback(formatter.EventHandlerFuncArray())
+			case utils.OutputModeJSON:
+				parser.SetEventCallback(printEventAsJSON)
+			}
+
+			// Gadgets with parser don't return anything, they provide the
+			// output via the parser
+			_, err = runtime.RunGadget(gadgetCtx)
+			if err != nil {
+				return fmt.Errorf("running gadget: %w", err)
+			}
+
+			return nil
+		},
+	}
+
+	if gadgetDesc.Type() != gadgets.TypeOneShot {
+		// Add timeout
+		cmd.PersistentFlags().IntVarP(&timeout, "timeout", "t", 0, "Number of seconds that the gadget will run for, 0 to disable")
+	}
+
+	cmd.PersistentFlags().BoolVarP(
+		&verbose,
+		"verbose", "v",
+		false,
+		"Print debug information",
+	)
+
+	outputFormats.Append(gadgets.OutputFormats{
+		"json": {
+			Name:        "JSON",
+			Description: "The output of the gadget is returned as raw JSON",
+			Transform:   nil,
+		},
+	})
+	defaultOutputFormat = "json"
+
+	// Add parser output flags
+	if parser != nil {
+		outputFormats.Append(gadgets.OutputFormats{
+			"columns": {
+				Name:        "Columns",
+				Description: "The output of the gadget is formatted in human readable columns",
+			},
+		})
+		defaultOutputFormat = "columns"
+
+		cmd.PersistentFlags().StringSliceVarP(
+			&showColumns,
+			"columns", "C",
+			parser.GetDefaultColumns(),
+			"Columns to output",
+		)
+		cmd.PersistentFlags().StringSliceVarP(
+			&filters,
+			"filter", "F",
+			[]string{},
+			"Filter rules",
+		)
+	}
+
+	// Add alternative output formats available in the gadgets
+	if outputFormatInterface, ok := gadgetDesc.(gadgets.GadgetOutputFormats); ok {
+		formats, defaultFormat := outputFormatInterface.OutputFormats()
+		outputFormats.Append(formats)
+		defaultOutputFormat = defaultFormat
+	}
+
+	outputFormatsHelp := buildOutputFormatsHelp(outputFormats)
+
+	cmd.PersistentFlags().StringVarP(
+		&outputMode,
+		"output",
+		"o",
+		defaultOutputFormat,
+		strings.Join(outputFormatsHelp, "\n")+"\n\n",
+	)
+
+	// Add params matching the gadget type
+	gadgetParams.Add(*gadgets.GadgetParams(gadgetDesc, parser).ToParams()...)
+
+	// Add gadget flags
+	addFlags(cmd, gadgetParams)
+
+	// Add per-gadget operator flags
+	for _, operatorParams := range operatorsParamCollection {
+		addFlags(cmd, operatorParams)
+	}
+	return cmd
+}
+
+func addFlags(cmd *cobra.Command, params *params.Params) {
+	for _, p := range *params {
+		desc := p.Description
+
+		if p.PossibleValues != nil {
+			desc += " [" + strings.Join(p.PossibleValues, ", ") + "]"
+		}
+
+		flag := cmd.PersistentFlags().VarPF(p, p.Key, p.Alias, desc)
+
+		// Allow passing a boolean flag as --foo instead of having to use --foo=true
+		if p.IsBoolFlag() {
+			flag.NoOptDefVal = "true"
+		}
+	}
+}
+
+func printEventAsJSON(ev any) {
+	d, err := json.Marshal(ev)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error marshalling %+v: %v", ev, err)
+		return
+	}
+	fmt.Fprintln(os.Stdout, string(d))
```

Comment:
The fronted should be used. Fell free to cherry-pick https://github.com/inspektor-gadget/inspektor-gadget/commit/d11e9a297a3a73453f9483015c7ec99fa0e3cd0f.